pkg/utils/gitrepo.go

```diff
@@ -999,6 +1006,44 @@ func matchUserSubAdmin(client client.Client, userIdentity, userGroups string) bo
 	return isUserSubAdmin
 }
 
+func scanUserSubAdmin(client client.Client, userIdentity, userGroups string) bool {
+	isUserSubAdmin := false
+
+	bindingList := &rbacv1.ClusterRoleBindingList{}
+
+	err := client.List(context.TODO(), bindingList)
+
+	if err == nil {
+		for _, binding := range bindingList.Items {
+			if binding.RoleRef.Kind == "ClusterRole" && binding.RoleRef.Name == "open-cluster-management:subscription-admin" {
+				klog.Infof("Found cluster role binding %s with open-cluster-management:subscription-admin cluster role.", binding.Name)
+
+				for _, subject := range binding.Subjects {
+					if strings.Trim(subject.Name, "") == strings.Trim(userIdentity, "") && strings.Trim(subject.Kind, "") == "User" {
+						klog.Info("User match. cluster-admin: true")
+
+						isUserSubAdmin = true
```

Comment:
Possible enhancement is to return true when we know the user is a sub admin and stop iterating through the loop